clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaConsumerServiceDelegator.java

```diff
@@ -25,9 +30,11 @@
  * that handling the writes before putting into the drainer queue is too expensive comparing to others.
  */
 public class KafkaConsumerServiceDelegator extends AbstractKafkaConsumerService {
-  private final KafkaConsumerService defaultConsumerService;
-  private final KafkaConsumerService consumerServiceForAAWCLeader;
   private final Function<String, Boolean> isAAWCStoreFunc;
+  // Map from a pair of <Version Topic, PubSubTopicPartition> to the consumer service assigned to it.
+  private final Map<Pair<PubSubTopic, PubSubTopicPartition>, KafkaConsumerService> vtTopicPartitionPairToConsumerService =
```

Comment:
https://github.com/linkedin/venice/blob/main/internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubTopicPartition.java#L7
`PubSubTopicPartition` has a reference to `PubSubTopic`, so do we still need to use `Pair` as key here?

Reply:
Oh, the pair here is mainly used for <version topic, version/real-time topic partition>, the reason is version topic could be different from real-time topic.

Reply:
Won't the referenced `PubSubTopic` contain the full name, so that we can differentiate whether it is a real-time topic or version topic?

Reply:
`PubSubTopic` inside `PubSubTopicPartition` could be real-time or version topic, the problem here is: our server host may ingest 2+ same real-time topic partitions from 2+ version topics (current/backup/future). As version topic + pubsubtopicpartition could be unique for that server host.

Reply:
Got it, make sense.